### PR TITLE
Use sirius namespacing for cmake

### PR DESCRIPTION
**Pull request by mtaillefumier**
SIRIUS build system use namespaces to avoid potential conflict with others packages build with cmake. It requires changes in the recipe that only apply to develop and the next official version of the package.

### CODE REVIEW

File: var/spack/repos/builtin/packages/sirius/package.py [hunk 1]
```diff
@@ -273,56 +273,64 @@ def libs(self):
...
         lapack = spec["lapack"]
         blas = spec["blas"]
 
         args.extend(
             [
-                self.define("LAPACK_FOUND", "true"),
-                self.define("LAPACK_LIBRARIES", lapack.libs.joined(";")),
-                self.define("BLAS_FOUND", "true"),
-                self.define("BLAS_LIBRARIES", blas.libs.joined(";")),
+                self.define(cm_label + "LAPACK_FOUND", "true"),
```
Comment (haampie) [inline var/spack/repos/builtin/packages/sirius/package.py]: Aren't these from `find_package(BLAS)`?

Comment (mtaillefumier) [reply var/spack/repos/builtin/packages/sirius/package.py]: Correction SIRIUS uses find(lapack) eventually.

Comment (mtaillefumier) [reply var/spack/repos/builtin/packages/sirius/package.py]: I will fix this

**Review by toxa81 — APPROVED** [review]
To mee it looks good, but experts @haampie have a last word

**Comment by mtaillefumier** [comment]
> To me it looks good, but experts @haampie have a last word

I agree I made the wrong assumption that SIRIUS had a custom find_lapack package. That's where my mistake comes from. It should be fixed with the latest commit

**Review by msimberg — APPROVED** [review]

File: var/spack/repos/builtin/packages/sirius/package.py (hunk 1 above)
Comment (mtaillefumier) [reply var/spack/repos/builtin/packages/sirius/package.py]: I reverted my changes after realizing that we do use the findLapack.cmake provided by cmake.

**Pull request merged**